setup.cfg

```diff
@@ -1,21 +1,21 @@
 [isort]
+profile = black
 skip =
     ./build,
     .nox,
     .tox,
     .scratch,
     _vendor,
     data
-multi_line_output = 3
 known_third_party =
     pip._vendor
 known_first_party =
     pip
     tests
 default_section = THIRDPARTY
-include_trailing_comma = true
 
 [flake8]
+max-line-length = 88
```

Comment:
@pradyunsg has anybody considered a more accessibility friendly value? Long lines are so painful for folks having large fonts (like me).

Reply:
88 is black's default. Like anything else around code style, it's personal preference and a trade-off. Shorter lines cause more line wrapping, which I hate more than the occasional long line. But I don't have a personal accessibility issue to consider.

Strong +1 on considering accessibility (I'm definitely *not* saying that I don't consider the problems of people who need large fonts as important) but is there any standard that suggests a different value is better overall for accessibility? I think this is something better agreed in a broader context, with pip following the lead of the consensus, rather than trying to make decisions independently.

Having said that, if there *is* any literature on the accessibility implications of line length choices for code style, I'd be interested in seeing it.

And just to be 100% clear, as a readability rule, I feel that shorter lines are better *anyway*, so I'd hope that regardless of this setting, most of pip's code would still be lines shorter than 88 characters, just because complex one-liners are bad.

Reply:
FWIW, I also have similar issues myself with extended line lengths and I think increasing it by 10% is fine by me. Fitting 80 wide files side-by-side on any of my devices is utterly out of question for me (except for the new fancy ultrawide) because of the font settings I use. This would arguably make that situation a bit worse by pushing closer to the limit of ~100 characters. ;)

black does provide a reference and justification for its choice tho, and mentions that values above 90 get too long.

https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length

Based off the bunch of reading I've had to do whenever folks randomly decide to make things too-wide on websites with no regard for accessibility (eg: virtualenv's docs) because "why waste space", I remember that ~90 characters/20-30 words is indeed the general consensus around prose as well. The exact numbers vary based on who *exactly* you ask but all of the sources I've found are 90 or above.

At the end of the day, this is something that comes down to being a trade-off and I'd really rather keep 88 here on the basis of "we're adopting black and all it's opinions here, because that's the black way".

Reply:
I would argue 80 is a better choice for pip specifically because the vast majority of the current code base uses 80. Whitespace reformatting diffs are frustrating to review (you try really hard to understand what changed, and realised nothing did) and using the default 88 would likely reformat too much lines in the code base to my liking. I don’t object to the 88 idea in a new project, but it’s better to stick with existing rules in a project with history like pip.

Reply:
> Whitespace reformatting diffs are frustrating to review (you try really hard to understand what changed, and realised nothing did)

Agreed. And you hit right at *why* they're annoying -- because you're trying to understand what changed.

These are dedicated code formatting PRs are basically 100% reformatting PRs, which does not need you to spend time to understand if/how the code logic changed -- because it hasn't -- and that changes the review process to be *much* less annoying.

> using the default 88 would likely reformat too much lines in the code base to my liking.

Fair enough.

---

At the end of the day, we're either configuring multiple tools to work with 80 (coz isort's black profile will need to be overriden) or just flake8 to work with black. :shrug:

Reply:
Nit: I just checked, it seems that the default line length for flake8 is actually 79, not 80.

> At the end of the day, we're either configuring multiple tools to work with 80 (coz isort's black profile will need to be overriden) or just flake8 to work with black. 🤷

Personally I explicitly configure all of them anyway for explicitness.

Reply:
Isn't this *precisely* the sort of debate that the "just use black" philosophy is intended to short circuit?

If arguing over line length is acceptable, can we have arguments over other choices black makes¹? Otherwise, doesn't the argument for using black boil down to "black lets us avoid any style debates that I don't feel like having" 😕 

¹ Banning single quotes!

Reply:
> Isn't this precisely the sort of debate that the "just use black" philosophy is intended to short circuit?

I'm very annoyed that we're even debating on the two knobs that black has. If only Łukasz had just picked 80 here, we could've skipped a knob (and a debate here)! :)

> Banning single quotes!

This is something I'm ambivalent on. I use `"` everywhere so, I personally don't feel like I'm ever "fighting" black. And, we basically have 0 consistency in the codebase today sooo... it's an improvement.

We can pass `--skip-string-normalization` and tell it to not change the quotes, if that makes you happier. :)

Reply:
> We can pass --skip-string-normalization and tell it to not change the quotes, if that makes you happier

No, let's just not debate line length instead :-)

Reply:
I don't have any literature about the specific line length. Although the reason I brought up accessibility is that I really use large fonts nowadays and, for example, when I'm in NeoVim (hence the dependency on the terminal setup) it only fits 72 chars for me (+ it consumes like 7 char-columns for the ruler/indicators on the left). I got into this after realizing that it's much less stressful for my eyes. But yes, I do understand how +10% may seem not that big of a deal for folks with smaller font habits. Sadly, many people are dismissive about this.

One argument I keep hearing is that now displays have high resolution (referring to the legacy TTY limitations) but they fail to take into account that (1) it's useful to have multiple vertical panes and (2) big displays are often not that bigger physically but have more pixels per inch. This basically means with the same pixel-based font size its actual physical size on screen would be smaller and that hurts accessibility (and so it encourages us to increase that size).

On a side note, as @pfmoore rightfully stated, shorter lines are easier to consume. And this is something to care about. Why is that? Code is read more often than it is written. But also *code is text*. And we *should* apply roughly the same principles to it too.
And what do we know about text in general? It's that hard media heavily utilize columns. This makes that text easier to consume because then you can read it vertically in line-wide chunks, almost instantly vs jumping left to right and back with your eyes all the time.
If you look at printed magazines, for example, their pages are wide, it is technically possible to print lines all the way from left to right but they choose to please the readers and make it less stressful to go through the text by actively utilizing columns with short lines.
Can you imagine how'd the UX aspect change if they chose otherwise?

FWIW here's what quick googling says on UX/accessibility, typography-wise (mostly web with references to printed media):
* https://www.dtelepathy.com/blog/products/increase-user-comprehension-readability-with-multi-column-text-layouts (web)
* https://accessibility.huit.harvard.edu/design-readability
* https://baymard.com/blog/line-length-readability
* https://ux.stackexchange.com/a/26363/47648
* https://useplaintext.email/#etiquette (recommends to apply line limits in emails too, I think I can be counted towards "web")

And these two also mention coding ergonomics specifically:
* https://softwareengineering.stackexchange.com/a/222998/75899
* https://sobolevn.me/2019/10/complexity-waterfall#lines
* https://blog.codinghorror.com/text-columns-how-long-is-too-long/

> black does provide a reference and justification for its choice tho

That justification comes off judgy or even, dare I say, insulting: it accuses/blames people for being paid by the line of code when all they want is for the thing to be readable...